packages/amplify-cli/src/context-manager.ts

```diff
@@ -4,29 +4,50 @@ import { PluginPlatform } from './domain/plugin-platform';
 import { attachExtentions } from './context-extensions';
 import { init } from './app-config';
 import { UsageData, NoUsageData } from './domain/amplify-usageData';
+import { ProjectSettings } from './domain/amplify-usageData/UsageDataPayload';
+import { stateManager } from 'amplify-cli-core';
 
 export function constructContext(pluginPlatform: PluginPlatform, input: Input): Context {
   const context = new Context(pluginPlatform, input);
   attachExtentions(context);
   return context;
 }
 
-export function attachUsageData(context: Context) {
+export async function attachUsageData(context: Context) {
   const { AMPLIFY_CLI_ENABLE_USAGE_DATA } = process.env;
   const config = init(context);
   const usageTrackingEnabled = AMPLIFY_CLI_ENABLE_USAGE_DATA
     ? AMPLIFY_CLI_ENABLE_USAGE_DATA === 'true'
     : config.usageDataConfig.isUsageTrackingEnabled;
+  const accountId = await context.amplify.executeProviderUtils(context, 'awscloudformation', 'getAccountId');
   if (usageTrackingEnabled) {
     context.usageData = UsageData.Instance;
   } else {
     context.usageData = NoUsageData.Instance;
   }
-  context.usageData.init(config.usageDataConfig.installationUuid, getVersion(context), context.input);
+  context.usageData.init(config.usageDataConfig.installationUuid, getVersion(context), context.input, accountId, getProjectSettings());
 }
 
 const getVersion = (context: Context) => context.pluginPlatform.plugins.core[0].packageVersion;
 
+const getProjectSettings = (): ProjectSettings => {
+  const projectSettings = {};
+  if (stateManager.projectConfigExists()) {
+    const projectConfig = stateManager.getProjectConfig();
+    const frontend = projectConfig['frontend'];
+    projectSettings['frontend'] = frontend;
+    if (projectConfig[frontend] && projectConfig[frontend].framework) {
```

Comment:
You can simplify this to `projectConfig?.[frontend].framework` and put it in a variable.